provider/metrics/container_insights.go

```diff
@@ -0,0 +1,315 @@
+		StartTime: pod.CreationTimestamp,
+		PodRef: stats.PodReference{
+			Name:      pod.Name,
+			Namespace: pod.Namespace,
+			UID:       string(pod.UID),
+		},
+		CPU: &stats.CPUStats{
+			UsageNanoCores:       newUInt64Pointer(0),
+			UsageCoreNanoSeconds: newUInt64Pointer(0),
+			Time:                 metav1.NewTime(time.Now()),
+		},
+		Memory: &stats.MemoryStats{
+			UsageBytes:      newUInt64Pointer(0),
+			WorkingSetBytes: newUInt64Pointer(0),
+			Time:            metav1.NewTime(time.Now()),
+		},
+		Network: &stats.NetworkStats{
+			Time: metav1.NewTime(time.Now()),
+			InterfaceStats: stats.InterfaceStats{
+				Name:     "eth0",
+				RxBytes:  newUInt64Pointer(0),
+				RxErrors: newUInt64Pointer(0),
+				TxBytes:  newUInt64Pointer(0),
+				TxErrors: newUInt64Pointer(0),
+			},
+		},
+		Containers: make([]stats.ContainerStats, 0),
+	}
+
+	for _, c := range pod.Status.ContainerStatuses {
+		containerName := c.Name
+		cs := stats.ContainerStats{
+			Name:      containerName,
+			StartTime: stat.StartTime,
+			CPU: &stats.CPUStats{
+				Time:                 metav1.NewTime(time.Now()),
+				UsageNanoCores:       newUInt64Pointer(0),
+				UsageCoreNanoSeconds: newUInt64Pointer(0),
+			},
+			Memory: &stats.MemoryStats{
+				Time:            metav1.NewTime(time.Now()),
+				RSSBytes:        newUInt64Pointer(0),
+				WorkingSetBytes: newUInt64Pointer(0),
+			},
+		}
+
+		if cpuSeries, found := metrics.getCPUMetricsSeries(containerName); found && len(cpuSeries) > 0 {
+			data := cpuSeries[len(cpuSeries)-1]
+			nanoCores := uint64(data.Average * 1000000)
+			cs.CPU.UsageNanoCores = &nanoCores
+			cs.CPU.Time = metav1.NewTime(data.Timestamp)
+
+			podCPUCore := *stat.CPU.UsageNanoCores
+			podCPUCore += nanoCores
+			stat.CPU.UsageNanoCores = &podCPUCore
+			stat.CPU.Time = metav1.NewTime(data.Timestamp)
+		}
+		if memorySeries, found := metrics.getMemoryMetricsSeries(containerName); found && len(memorySeries) > 0 {
+			data := memorySeries[len(memorySeries)-1]
+			bytes := uint64(data.Average)
+			cs.Memory.UsageBytes = &bytes
+			cs.Memory.WorkingSetBytes = &bytes
+			cs.Memory.Time = metav1.NewTime(data.Timestamp)
+
+			podMem := *stat.Memory.UsageBytes
+			podMem += bytes
+			stat.Memory.UsageBytes = &podMem
+			stat.Memory.WorkingSetBytes = &podMem
+			stat.Memory.Time = metav1.NewTime(data.Timestamp)
+		}
+		stat.Containers = append(stat.Containers, cs)
+	}
+
+	return stat
+}
+
+func (containerInsights *containerInsightsPodStatsGetter) updateCumulativeValues(logger log.Logger, pod *v1.Pod, metrics *containerInsightsMetricsWrapper, stat *stats.PodStats) {
+	cacheKey := string(pod.UID)
+	c, found := containerInsights.cumulativeUsageCache.Get(cacheKey)
+	var cachedCumulativeValue *podCumulativeUsage
+	if !found {
+		logger.Infof("Didn't find cumulative values in cache for pod: pod name '%s', UID '%s'", pod.Name, cacheKey)
+		cachedCumulativeValue = &podCumulativeUsage{
+			podUID: cacheKey,
+			networkRx: cumulativeUsage{
```

Comment:
These initializations are redundant.

Reply:
I did that specifically although I know that is redundant. Because:
1. Make sure every filed be explicitly initialized, instead of rely on Go's default value. 
2. That will be easier for reading codes, especially for the reader without too much knowledge about the Golang. Actually I had same issue when I had not too much Golang knowledge.